### PR TITLE
[#101129996] Add nats: AWS PR#1

### PR DESCRIPTION
**Pull request by mtekel**
https://www.pivotaltracker.com/n/projects/1275640/stories/101129996

First PR for aws - preparing the environment for CF NATS template:
- stop using EIP for microbosh (unnecessary)
- adjust instances (disk sizes, instance types)
- create subnet for NATS
- continue provisioning:
  - install bosh cli
  - upload stemcell to microbosh
  - upload v215 release to microbosh

**TESTING**
`make aws DEPLOY_ENV=your_environment`
Watch it provision. Be patient, it takes about 30 mins now. There's potential for considerable speedups.
When done, ssh to bastion host. Do:
- `bosh status` - you should get response from microbosh like 

```
Config
             /home/ubuntu/.bosh_config

Director
  Name       my-bosh
  URL        https://10.0.0.6:25555
  Version    1.3029.0 (00000000)
  User       admin
  UUID       e7004343-b5a3-4f78-b531-f472cce9046e
  CPI        cpi
  dns        disabled
  compiled_package_cache disabled
  snapshots  disabled

Deployment
  not set
```
- `bosh releases` should return

```
Acting as user 'admin' on 'my-bosh'

+------+----------+-------------+
| Name | Versions | Commit Hash |
+------+----------+-------------+
| cf   | 215      | 2a4897df+   |
+------+----------+-------------+
(+) Uncommitted changes

Releases total: 1
```
- `bosh stemcells` should give you

```
Acting as user 'admin' on 'my-bosh'

+-------------------------------------+---------------+---------+--------------+
| Name                                | OS            | Version | CID          |
+-------------------------------------+---------------+---------+--------------+
| bosh-aws-xen-ubuntu-trusty-go_agent | ubuntu-trusty | 3056    | ami-1690cb61 |
+-------------------------------------+---------------+---------+--------------+

(*) Currently in-use

Stemcells total: 1
```

**Reviewing**
Anyone but @mtekel or @RichardKnop

### CODE REVIEW

File: aws/provision.sh [hunk 1]
```diff
@@ -14,6 +14,35 @@ chmod 400 ~/.ssh/id_rsa.pub
 eval `ssh-agent`
 ssh-add ~/.ssh/id_rsa
 
+# TODO: download bosh-init from our own bucket
```
Comment (dcarley) [inline aws/provision.sh:17]: Can you explain why? Do they remove old releases from the official source?

File: aws/provision.sh [hunk 2]
```diff
@@ -14,6 +14,35 @@ chmod 400 ~/.ssh/id_rsa.pub
...
+export BOSH_INIT_LOG_LEVEL=debug
+export BOSH_INIT_LOG_PATH=bosh_init.log
+time ./bosh-init deploy manifest_aws.yml
+
+# TODO: install bosh cli rubygems in parallel to bosh-init (which mainly stresses bosh server)
+echo "Installing bosh cli..."
+time sudo gem install bosh_cli -v 1.3056.0 --no-ri --no-rdoc
+export PATH=$PATH:/usr/local/bin/bosh
+
+# FIXME: make passwords properly strong
```
Comment (dcarley) [inline aws/provision.sh:29]: We should create a story for this.

File: aws/provision.sh [hunk 3]
```diff
@@ -14,6 +14,35 @@ chmod 400 ~/.ssh/id_rsa.pub
...
 
+# TODO: download bosh-init from our own bucket
 wget https://s3.amazonaws.com/bosh-init-artifacts/bosh-init-0.0.72-linux-amd64 -O bosh-init
 chmod +x bosh-init
-./bosh-init deploy manifest_aws.yml
+export BOSH_INIT_LOG_LEVEL=debug
+export BOSH_INIT_LOG_PATH=bosh_init.log
+time ./bosh-init deploy manifest_aws.yml
+
+# TODO: install bosh cli rubygems in parallel to bosh-init (which mainly stresses bosh server)
```
Comment (dcarley) [inline aws/provision.sh:24]: Should we have one or more stories of ways to speed up the install process?

File: aws/provision.sh (hunk 1 above)
Comment (mtekel) [inline aws/provision.sh:17]: Slow & they might remove it. It's better to manage this yourself where you have full control.

File: aws/provision.sh (hunk 3 above)
Comment (mtekel) [inline aws/provision.sh:24]: Yes, contacted Brett already.

**Comment by mtekel** [comment]
After looking at GCE side, more work is needed on AWS as well to unify them both, closing this PR.

**Pull request closed without merging**